streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -292,21 +278,36 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
 
             while (offset < highWatermark) {
                 try {
-                    final ConsumerRecords<byte[], byte[]> records = globalConsumer.poll(pollTime);
+                    final ConsumerRecords<byte[], byte[]> records =
+                        globalConsumer.poll(pollTimePlusRequestTimeoutPlusTaskTimeout);
```

Comment:
It does seem a bit weird here to add in the request timeout. Not sure I follow the reasoning behind that...

Reply:
Unfortunately, Github ate the very extensive thread about this: https://github.com/apache/kafka/pull/9047/files/0e07109bb0cfe87c85e76fbab3b50e9274300388#r460449901